honeybee_radiance/modifierset.py

```diff
@@ -0,0 +1,702 @@
+        """Get a modifier object that will be assigned to a given type of aperture.
+
+        Args:
+            boundary_condition: Text string for the boundary condition
+                (eg. 'Outdoors', 'Surface')
+            aperture_type: Text string for the type of aperture
+                (eg. 'Window', 'OperableWindow', 'GlassDoor').
+            parent_face_type: Text string for the type of face to which the aperture
+                is a child (eg. 'Wall', 'Floor', 'Roof').
+        """
+        if boundary_condition == 'Outdoors':
+            if aperture_type in ('Window', 'OperableWindow'):
+                if parent_face_type == 'Wall':
+                    return self.aperture_set.exterior_window_modifier
+                else:
+                    return self.aperture_set.skylight_modifier
+            elif aperture_type == 'GlassDoor':
+                return self.aperture_set.glass_door_modifier
+            else:
+                raise NotImplementedError('Aperture type {} is not recognized for '
+                                          'ModifierSet'.format(aperture_type))
+        elif boundary_condition == 'Surface':
+            return self.aperture_set.interior_modifier
+        else:
+            raise NotImplementedError(
+                'Boundary condition {} is not recognized for apertures in '
+                'ModifierSet'.format(boundary_condition))
+
+    def get_door_modifier(self, boundary_condition, parent_face_type):
+        """Get a modifier object that will be assigned to a given type of door.
+
+        Args:
+            boundary_condition: Text string for the boundary condition
+                (eg. 'Outdoors', 'Surface')
+            parent_face_type: Text string for the type of face to which the door
+                is a child (eg. 'Wall', 'Floor', 'Roof').
+        """
+        if boundary_condition == 'Outdoors':
+            if parent_face_type == 'Wall':
+                return self.door_set.exterior_modifier
+            else:
+                return self.door_set.overhead_modifier
+        elif boundary_condition == 'Surface':
+            return self.door_set.interior_modifier
+        else:
+            raise NotImplementedError(
+                'Boundary condition {} is not recognized for doors in '
+                'ModifierSet'.format(boundary_condition)
+                )
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a ModifierSet from a dictionary.
+
+        Note that the dictionary must be a non-abridged version for this
+        classmethod to work.
+
+        Args:
+            data: Dictionary describing the ModifierSet.
+        """
+        assert data['type'] == 'ModifierSet', \
+            'Expected ModifierSet. Got {}.'.format(data['type'])
+
+        # TODO: build each modifier from JSON
+        # TODO: put each set together
+        # NOTE: An alternative solution is to implement from_dict to sets
+        # and use that to recreate the class.
```

Comment:
I would say that you should feel free to go for this alternative here.  The only reason why it did not make sense in honeybee-energy is because you often have a lot of materials shared between all of the constructions across the set.  You don't have the same restriction here and, while there might be a few duplicated modifiers across the whole set, I don't think that you are saving enough duplication to merit a solution like that in `honeybee-energy`.